llvm_mode/afl-llvm-pass.so.cc

```diff
@@ -116,14 +116,24 @@ bool AFLCoverage::runOnModule(Module &M) {
   }
   
   for (auto &F : M){
-    for (auto &BB : F) {
 
-      assert(block_counter < MAP_SIZE && "counter is too large");
+    for (auto &BB : F) {
       BasicBlock::iterator IP = BB.getFirstInsertionPt();
       IRBuilder<> IRB(&(*IP));
 
       /* Make up cur_loc based on block counter */
 
+      MDNode* BBid_meta;
+      for (Instruction& instr : BB.getInstList()) {
+            assert(instr.hasMetadata() && "The first instruction of the block should include block id, have you instrumented the code with set-counter-BBid-llvm-pass first?");
+            BBid_meta = instr.getMetadata("BBid");
+            break;
+      }
+      char c;
+      std::string meta_string = cast<MDString>(BBid_meta->getOperand(0))->getString();
+      std::sscanf(meta_string.c_str(), "%" SCNd64 "%c", &block_counter, &c);
```

Comment:
sscanf does not detect overflow. In C, we should use [strtoull](https://linux.die.net/man/3/strtoull) with the right checks. Here's an [example](https://github.com/Samsung/afl_cc/blob/master/afl-fuzz.c#L1959), which I encourage you to check if it's correct...